src/profile-logic/call-node-info.js

```diff
@@ -274,51 +270,298 @@ export class CallNodeInfoImpl implements CallNodeInfo {
+  func: IndexIntoFuncTable[], // IndexIntoInvertedNonRootCallNodeTable -> IndexIntoFuncTable
+  pathHash: string[], // IndexIntoInvertedNonRootCallNodeTable -> string
+  category: IndexIntoCategoryList[], // IndexIntoInvertedNonRootCallNodeTable -> IndexIntoCategoryList
+  subcategory: IndexIntoSubcategoryListForCategory[], // IndexIntoInvertedNonRootCallNodeTable -> IndexIntoSubcategoryListForCategory
+  innerWindowID: InnerWindowID[], // IndexIntoInvertedNonRootCallNodeTable -> InnerWindowID
+  // IndexIntoNativeSymbolTable: all frames that collapsed into this call node inlined into the same native symbol
+  // -1: divergent: some, but not all, frames that collapsed into this call node were inlined, or they are from different symbols
+  // null: no inlining
+  sourceFramesInlinedIntoSymbol: Array<IndexIntoNativeSymbolTable | -1 | null>,
+  suffixOrderIndexRangeStart: SuffixOrderIndex[], // IndexIntoInvertedNonRootCallNodeTable -> SuffixOrderIndex
+  suffixOrderIndexRangeEnd: SuffixOrderIndex[], // IndexIntoInvertedNonRootCallNodeTable -> SuffixOrderIndex
+
+  // Non-null for non-root nodes whose children haven't been created yet.
+  // For a non-root node x of the inverted tree, let k = depth[x] its depth in the inverted tree,
+  // and deepNodes = deepNodesForSuffixOrderIndexRange[x] be its non-null deep nodes.
+  // Then, for every index i in suffixOrderIndexRangeStart[x]..suffixOrderIndexRangeEnd[x],
+  // the k'th prefix node of suffixOrderedCallNodes[i] is stored at deepNodes[i - suffixOrderIndexRangeStart[x]].
+  deepNodesForSuffixOrderIndexRange: Array<Uint32Array | null>, // IndexIntoInvertedNonRootCallNodeTable -> (Uint32Array | null)
+
+  depth: number[], // IndexIntoInvertedNonRootCallNodeTable -> number
+  length: number,
+|};
+
+// Compute the InvertedRootCallNodeTable.
+// We compute this information upfront for all roots. The root count is fixed -
+// the number of roots is the same as the number of functions in the funcTable.
+function _createInvertedRootCallNodeTable(
+  callNodeTable: CallNodeTable,
+  rootSuffixOrderIndexRangeEndCol: Uint32Array,
+  suffixOrderedCallNodes: Uint32Array,
+  defaultCategory: IndexIntoCategoryList
+): InvertedRootCallNodeTable {
+  const funcCount = rootSuffixOrderIndexRangeEndCol.length;
+  const category = new Int32Array(funcCount);
+  const subcategory = new Int32Array(funcCount);
+  const innerWindowID = new Float64Array(funcCount);
+  const sourceFramesInlinedIntoSymbol = new Array(funcCount);
+  let previousRootSuffixOrderIndexRangeEnd = 0;
+  for (let funcIndex = 0; funcIndex < funcCount; funcIndex++) {
+    const callNodesuffixOrderIndexRangeStart =
+      previousRootSuffixOrderIndexRangeEnd;
+    const callNodesuffixOrderIndexRangeEnd =
+      rootSuffixOrderIndexRangeEndCol[funcIndex];
+    previousRootSuffixOrderIndexRangeEnd = callNodesuffixOrderIndexRangeEnd;
+    if (
+      callNodesuffixOrderIndexRangeStart === callNodesuffixOrderIndexRangeEnd
+    ) {
```

Comment:
That's right. Changed it to the following:

```js
    if (
      callNodeSuffixOrderIndexRangeStart === callNodeSuffixOrderIndexRangeEnd
    ) {
      // This root is never actually displayed in the inverted tree. It
      // corresponds to a func which has no self time - no non-inverted node has
      // this func as its self func. This root only exists for simplicity, so
      // that there is one root per func.

      // Set all columns to zero / null for this root.
      sourceFramesInlinedIntoSymbol[funcIndex] = null;
      // (the other columns are already initialized to zero because they're
      // typed arrays)
      continue;
    }
```